src/db/commit.rs

```diff
@@ -7,6 +7,40 @@ use str_macro::str;
 
 pub const DEFAULT_HEAD_NAME: &str = "main";
 
+#[derive(Clone, Copy)]
+enum Ref<'a> {
+    None,
+    Strong(&'a str),
+    Weak(&'a str),
+}
+
+impl<'a> Ref<'a> {
```

Comment:
One more thought: it makes me a little nervous that whether something is a strong or weak ref is not tied to whether the thing is in the refs or not. Seems like we want one to imply the other.

Reply:
It did imply that but it is all inside `new_impl`